### PR TITLE
fix: approve session proposal, receive session disconnect

### PR DESCRIPTION
**Pull request by jakubuid**

### CODE REVIEW

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/domain/SignEngine.kt [hunk 1]
```diff
@@ -358,6 +368,6 @@ internal class SignEngine(
     }
 
     companion object {
-        private const val WATCHER_INTERVAL = 3000L
+        private const val WATCHER_INTERVAL = 5000L
```
Comment (TalhaAli00) [inline protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/domain/SignEngine.kt:371]: what do you think about bumping this value to something like 10 or even 30 sec? I think to save on resources, we could check less frequently. We also have other checks to validate expiry so I think we would be ok having something longer

File: core/android/src/main/kotlin/com/walletconnect/android/pairing/engine/domain/PairingEngine.kt [hunk 2]
```diff
@@ -317,9 +341,9 @@ internal class PairingEngine(
                 }
             }.launchIn(scope)
 
-    private fun resubscribeToPairingFlow() {
+    private fun resubscribeToPairing() {
         try {
-            val pairingTopics = pairingRepository.getListOfPairings().filter { pairing -> pairing.isNotExpired() }.map { pairing -> pairing.topic.value }
+            val pairingTopics = runBlocking { pairingRepository.getListOfPairings().filter { pairing -> pairing.isNotExpired() }.map { pairing -> pairing.topic.value } }
```
Comment (TalhaAli00) [inline core/android/src/main/kotlin/com/walletconnect/android/pairing/engine/domain/PairingEngine.kt:346]: why does this need to be blocking? I think we should think about going the other way and launch a coroutine in the init block and make this function a suspend function to run async. We're already launching coroutines inside of this function

**Review by TalhaAli00 — APPROVED** [review]
also curious about why you switched the toast code in the sample

**Comment by jakubuid** [comment]
> also curious about why you switched the toast code in the sample

It was causing session request and proposal dialogs to disappear in some corner cases

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/domain/SignEngine.kt (hunk 1 above)
Comment (jakubuid) [reply protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/domain/SignEngine.kt:371]: Yeah I agree. I think that checking every 30s is fine

File: core/android/src/main/kotlin/com/walletconnect/android/pairing/engine/domain/PairingEngine.kt (hunk 2 above)
Comment (jakubuid) [reply core/android/src/main/kotlin/com/walletconnect/android/pairing/engine/domain/PairingEngine.kt:346]: Yeah, good point. Made this function a suspend function.

**Pull request merged**